packages/next/src/bin/next.ts

```diff
@@ -68,75 +23,324 @@ if (
+        if (isNotStandard || shouldWarnCommands.includes(cmdName)) {
+          warn(NON_STANDARD_NODE_ENV)
+        }
+      }
+
+      ;(process.env as any).NODE_ENV = process.env.NODE_ENV || defaultEnv
+      ;(process.env as any).NEXT_RUNTIME = 'nodejs'
 
-if (process.env.NODE_ENV) {
-  const isNotStandard = !standardEnv.includes(process.env.NODE_ENV)
-  const shouldWarnCommands =
-    process.env.NODE_ENV === 'development'
-      ? ['start', 'build']
-      : process.env.NODE_ENV === 'production'
-      ? ['dev']
-      : []
+      if (thisCommand.getOptionValue('inspect') === true) {
+        console.error(
+          `\`--inspect\` flag is deprecated. Use env variable NODE_OPTIONS instead: NODE_OPTIONS='--inspect' next ${cmdName}`
+        )
+        process.exit(1)
+      }
+    })
 
-  if (isNotStandard || shouldWarnCommands.includes(command)) {
-    log.warn(NON_STANDARD_NODE_ENV)
+    return cmd
   }
 }
 
-;(process.env as any).NODE_ENV = process.env.NODE_ENV || defaultEnv
-;(process.env as any).NEXT_RUNTIME = 'nodejs'
+const program = new MyRootCommand()
 
-if (command === 'build') {
-  process.on('SIGTERM', () => process.exit(0))
-  process.on('SIGINT', () => process.exit(0))
-}
+program
+  .name('next')
+  .description(
+    'The Next.js CLI allows you to develop, build, start your application, and more.'
+  )
+  .configureHelp({
+    formatHelp: (cmd, helper) => formatCliHelpOutput(cmd, helper),
+    subcommandTerm: (cmd) => `${cmd.name()} ${cmd.usage()}`,
+  })
+  .helpCommand(false)
+  .helpOption('-h, --help', 'Displays this message.')
+  .version(
+    `Next.js v${process.env.__NEXT_VERSION}`,
+    '-v, --version',
+    'Outputs the Next.js version.'
+  )
 
-async function main() {
-  const currentArgsSpec = commandArgs[command]()
-  const validatedArgs = getValidatedArgs(currentArgsSpec, forwardedArgs)
-
-  for (const dependency of ['react', 'react-dom']) {
-    try {
-      // When 'npm link' is used it checks the clone location. Not the project.
-      require.resolve(dependency)
-    } catch (err) {
-      console.warn(
-        `The module '${dependency}' was not found. Next.js requires that you include it in 'dependencies' of your 'package.json'. To add it, run 'npm install ${dependency}'`
-      )
-    }
-  }
+program
+  .command('build')
+  .description(
+    'Creates an optimized production build of your application. The output displays information about each route.'
+  )
+  .argument(
+    '[directory]',
+    `A directory on which to build the application. ${italic(
+      'If no directory is provided, the current directory will be used.'
+    )}`
+  )
+  .option('-d, --debug', 'Enables a more verbose build output.')
+  .option('--profile', 'Enables production profiling for React.')
+  .option('--no-lint', 'Disables linting.')
+  .option('--no-mangling', 'Disables mangling.')
+  .addOption(
+    new Option(
+      '--experimental-app-only',
+      'Builds only App Router routes.'
+    ).hideHelp()
```

Comment:
Let's keep this in the help output like it was before

Reply:
<picture data-single-emoji=":done:" title=":done:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/done/556da3823bdfab37.png" alt=":done:" align="absmiddle"></picture>